dds/DCPS/RTPS/ParameterListConverter.cpp

```diff
@@ -36,6 +36,39 @@ namespace {
     param_list[length] = param;
   }
 
+  void extract_type_info_param(const Parameter& param, XTypes::TypeInformation& type_info, bool swap_bytes) {
+    ACE_Data_Block db(param.type_information().length(), ACE_Message_Block::MB_DATA,
+          reinterpret_cast<const char*>(param.type_information().get_buffer()),
+          0 /*alloc*/, 0 /*lock*/, ACE_Message_Block::DONT_DELETE, 0 /*db_alloc*/);
+    ACE_Message_Block data(&db, ACE_Message_Block::DONT_DELETE, 0 /*mb_alloc*/);
+          data.wr_ptr(data.space());
+    OpenDDS::DCPS::Serializer serializer(&data, swap_bytes);
```

Comment:
A few things about this:
- You will need to enable XCDR2 behavior from `Serializer` itself by giving it a XCDR2 `Encoding` or `Encoding::Kind`.
- Right now I'm planning on removing this constructor from `Serializer`.
- I think the RTPS parameter lists that this is going into are always little endian.

```suggestion
    DCPS::Serializer serializer(&data,
      DCPS::Encoding::KIND_XCDR2_PARAMLIST, DCPS::LITTLE_ENDIAN);
```

Reply:
>     * I think the RTPS parameter lists that this is going into are always little endian.

I think this is related to the swap_bytes comment I made before.  I agree that the current implementation is using PL_CDR_LE (see the `Serializer` object in `Sedp::Writer::write_parameter_list`).  We could make this a constant in `MessageTypes.h` so we're not just depending on an implementation detail.